### ISSUE TITLE
TypeScript type for EntityDTO with `ManyToOne` field is not correct

### ISSUE THREAD

**Issue by dzcpy** [mikro-orm/mikro-orm#2414]
**Describe the bug**
When JSONify entities which has unpopulated ManyToOne fields, the return result has wrong TypeScript types.

**Stack trace**

**To Reproduce**
1. Given the following entities:
```ts
@Entity()
export class TeamMember extends BaseEntity<TeamMember, 'id'> {
  @PrimaryKey({ type: 'uuid', defaultRaw: 'uuid_generate_v4()' })
  id: string;

  @ManyToOne({ entity: () => Team, onDelete: 'CASCADE', onUpdateIntegrity: 'CASCADE' })
  team: IdentifiedReference<Team>;
}
```

```ts
@Entity()
export class Team extends BaseEntity<Team, 'id'> {
  @PrimaryKey({ type: 'uuid', defaultRaw: 'uuid_generate_v4()' })
  id: string;

  @OneToMany({ entity: () => TeamMember, mappedBy: (teamMember) => teamMember.team, orphanRemoval: true, cascade: [Cascade.ALL] })
  members = new Collection<TeamMember>(this);
}
```
2. Get the JSON verson of TeamMember record:
```ts
const teamMember = await orm.em.findOne(TeamMember, '{actual-team-member-id}');
const teamMemberJSON: EntityDTO<TeamMember> = teamMember.toJSON();
console.log(teamMemberJSON.team);
// 022b5aab-4a04-4c86-a9f0-e6c9748a7f74
```
`teamMemberJSON.team`'s type now should be `EntityDTO<Team>`, but actually, it's a string id. So it causes some type errors

**Expected behavior**
Either set unpopulatd ManyToOne fields of JSONified object to string, or something like `team: { id: '{actual-id}' }`;

**Additional context**

**Versions**

| Dependency | Version |
| - | - |
| node | 17 |
| typescript | 4.4.4 |
| mikro-orm | 5.0.0-dev.439 |
| your-driver | postgresql |

**Comment by B4nan** [comment]
It can be PK or an object, depending of if you populate it. This information is available on the `Loaded` type, so even if we would have the `EntityDTO` type supporting it, you would basically erase it by explicitly defining your `teamMemberJSON` variable type (here it does not matter as you are not populating the relation). 

Don't expec too much magic, those types will be always a bit wonky as we don't have the full context in many places. But in this particular example I hope we can handle it.

**Comment by B4nan** [comment]
This will be finally resolved with the TS 4.8 update, including the inference for base entity methods. It will be slightly breaking (on type level) release for those who define custom `toJSON` methods - the signature will need to return `EntityDTO<this>` instead of `EntityDTO<User>` etc.
